src/mongocrypt-ctx-private.h

```diff
@@ -137,6 +137,10 @@ typedef struct {
    _mongocrypt_buffer_t key_doc;
    _mongocrypt_buffer_t plaintext_key_material;
    _mongocrypt_buffer_t encrypted_key_material;
+
+   char *kmip_unique_identifier;
```

Comment:
I find it much preferable to cast on free rather than permitting users unnecessary permissions to modify the target string. `free` (and equivalents) often not taking pointer-to-const is indeed unfortunate for reasons outlined in the linked SO post.